liquibase-core/src/main/java/liquibase/change/core/LoadDataChange.java

```diff
@@ -884,6 +829,90 @@ public String getSerializedObjectNamespace() {
         return STANDARD_CHANGELOG_NAMESPACE;
     }
 
+
+    protected SqlStatement[] generateStatementsFromRows(Database database, List<LoadDataRowConfig> rows) {
+        List<SqlStatement> statements = new ArrayList<>();
+        List<ExecutablePreparedStatementBase> preparedStatements = new ArrayList<>();
+
+        for (LoadDataRowConfig row : rows) {
+            List<LoadDataColumnConfig> columnsFromCsv = row.getColumns();
+            if (row.needsPreparedStatement()) {
+                ExecutablePreparedStatementBase stmt =
+                        this.createPreparedStatement(
+                                database, getCatalogName(), getSchemaName(), getTableName(), columnsFromCsv,
+                                getChangeSet(), Scope.getCurrentScope().getResourceAccessor()
+                        );
+                preparedStatements.add(stmt);
+            } else {
+                InsertStatement insertStatement =
+                        this.createStatement(getCatalogName(), getSchemaName(), getTableName());
+
+                for (LoadDataColumnConfig column : columnsFromCsv) {
+                    String columnName = column.getName();
+                    Object value = column.getValueObject();
+
+                    if (value == null) {
+                        value = "NULL";
+                    }
+
+                    insertStatement.addColumnValue(columnName, value);
+
+                    if (insertStatement instanceof InsertOrUpdateStatement) {
+                        ((InsertOrUpdateStatement) insertStatement).setAllowColumnUpdate(columnName, column.getAllowUpdate() == null || column.getAllowUpdate());
+                    }
+                }
+
+                statements.add(insertStatement);
+            }
+        }
+        if (rows.stream().anyMatch(LoadDataRowConfig::needsPreparedStatement)) {
+            // If we have only prepared statements and the database supports batching, let's roll
```

Comment:
I think the refactoring exhibits a bit more clearly a bug that was already there.

"any prepared statements" gets conflated with "only prepared statements". As a result, regular statements will be skipped as soon as 1 prepared statement is created.